src/server/application.rs

```diff
@@ -244,137 +244,168 @@ impl SwayOSDApplication {
 		self.app.run().into()
 	}
 
+	fn choose_windows(osd_app: &SwayOSDApplication) -> Vec<SwayosdWindow> {
+		let mut selected_windows = Vec::new();
+
+		match get_monitor_name() {
+			Some(monitor_name) => {
+				for window in osd_app.windows.borrow().to_owned() {
+					if let Some(monitor_connector) = window.monitor.connector() {
+						if monitor_name == monitor_connector {
+							selected_windows.push(window);
+						}
+					}
+				}
+			}
+			None => return osd_app.windows.borrow().to_owned(),
+		}
+
```

Comment:
Fails to display when an unknown output name is provided. Returning all the windows when the vector is empty would fix this :)